climada/engine/unsequa/calc_cost_benefit.py

```diff
@@ -165,55 +178,47 @@ def uncertainty(self, unc_data, pool=None, **cost_benefit_kwargs):
+        Parallelization logic is described in the base class
+        here :py:class:`~climada.engine.unsequa.calc_base.Calc`
+
         See Also
         --------
         climada.engine.cost_benefit:
-            Compute risk and adptation option cost benefits.
+            compute risk and adptation option cost benefits.
 
         """
 
-        if unc_data.samples_df.empty:
+        if unc_sample.samples_df.empty:
             raise ValueError("No sample was found. Please create one first" +
                         "using UncImpact.make_sample(N)")
 
-        samples_df = unc_data.samples_df.copy(deep=True)
+        '''copy may not be needed, but is kept to prevent potential
+        data corruption issues. The computational cost should be
+        minimal as only a list of floats is copied.'''
+        samples_df = unc_sample.samples_df.copy(deep=True)
+
+        if chunksize is None:
+            chunksize = _multiprocess_chunksize(samples_df, processes)
         unit = self.value_unit
 
         LOGGER.info("The freq_curve is not saved. Please "
                     "change the risk_func (see climada.engine.cost_benefit) "
                     "if return period information is needed")
 
+        one_sample = samples_df.iloc[0:1]
         start = time.time()
-        one_sample = samples_df.iloc[0:1].iterrows()
-        cb_metrics = map(self._map_costben_calc, one_sample)
-        [imp_meas_present,
-         imp_meas_future,
-         tot_climate_risk,
-         benefit,
-         cost_ben_ratio] = list(zip(*cb_metrics))
+        self._compute_cb_metrics(one_sample, cost_benefit_kwargs, chunksize=1, processes=1)
         elapsed_time = (time.time() - start)
-        self.est_comp_time(unc_data.n_samples, elapsed_time, pool)
+        self.est_comp_time(unc_sample.n_samples, elapsed_time, processes)
 
         #Compute impact distributions
-        with log_level(level='ERROR', name_prefix='climada'):
-            if pool:
-                LOGGER.info('Using %s CPUs.', pool.ncpus)
-                chunksize = max(min(unc_data.n_samples // pool.ncpus, 100), 1)
-                cb_metrics = pool.map(partial(self._map_costben_calc, **cost_benefit_kwargs),
-                                      samples_df.iterrows(),
-                                      chunksize=chunksize)
-
-            else:
-                cb_metrics = map(partial(self._map_costben_calc, **cost_benefit_kwargs),
-                                 samples_df.iterrows())
-
-        #Perform the actual computation
-        with log_level(level='ERROR', name_prefix='climada'):
-            [imp_meas_present,
-             imp_meas_future,
-             tot_climate_risk,
-             benefit,
-             cost_ben_ratio] = list(zip(*cb_metrics)) #Transpose list of list
+        [imp_meas_present,
+        imp_meas_future,
+        tot_climate_risk,
+        benefit,
+        cost_ben_ratio] = self._compute_cb_metrics(samples_df, cost_benefit_kwargs, chunksize, processes)
```

Comment:
Line too long. Black formats it like this:
```suggestion
        [
            imp_meas_present,
            imp_meas_future,
            tot_climate_risk,
            benefit,
            cost_ben_ratio,
        ] = self._compute_cb_metrics(
            samples_df, cost_benefit_kwargs, chunksize, processes
        )
```